operator/pkg/controllers/configconnectorcontext/configconnectorcontext_controller_test.go

```diff
@@ -1377,3 +1380,33 @@ func newConfigConnectorContextReconcilerWithCustomizationWatcher(m ctrl.Manager)
 		})
 	return r
 }
+
+func TestControllerOverridesField(t *testing.T) {
+	t.Parallel()
+	ccc := &corev1beta1.ConfigConnectorContext{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      corev1beta1.ConfigConnectorContextAllowedName,
+			Namespace: "foo-ns",
+		},
+		Spec: corev1beta1.ConfigConnectorContextSpec{
+			Experiments: &corev1beta1.Experiments{
+				ControllerOverrides: map[string]k8sreconciler.ReconcilerType{
+					"BigQueryDataset.bigquery.cnrm.cloud.google.com": "direct",
```

Comment:
Can we add tests for the following?
1) 
<valid resource 1> : <valid controller A> 
<valid resource 2> : <valid controller A> 
2)
<valid resource 1> : <valid controller A> 
<valid resource 1> : <valid controller B>
3) 
<valid resource> : <invalid controller>
4)
<invalid resource> : <valid controller>
5)
<invalid resource> : <invalid controller>
6) 
<valid resource 1> : <default controller for resource 1>
7) 
<valid resource 1> : <valid controller which is not legal for resource 1>